src/decisionengine/framework/modules/tests/test_de_logger.py

```diff
@@ -1,49 +1,64 @@
-import logging
+import gc
 import tempfile
-import unittest
 
 import pytest
+import structlog
 
 import decisionengine.framework.modules.de_logger as de_logger
 
-class TestLogger(unittest.TestCase):
-
-    def setUp(self):
-        mylogger = logging.getLogger("decision_engine")
-        while mylogger.hasHandlers():
-            try:
-                mylogger.removeHandler(mylogger.handlers[0])
-            except Exception:
-                break
-
-    def test_by_nonsense_is_err(self):
-        with pytest.raises(ValueError) as err:
-            with tempfile.NamedTemporaryFile() as log:
-                de_logger.set_logging(log_level='INFO', max_backup_count=6,
-                                      file_rotate_by="nonsense", max_file_size=1000000,
-                                      log_file_name=log.name)
-        assert "Incorrect 'file_rotate_by'" in str(err.value)
-
-    def test_by_size(self):
-        with tempfile.NamedTemporaryFile() as log:
-            mylogger = de_logger.set_logging(log_level='INFO', max_backup_count=6,
-                                             file_rotate_by="size", max_file_size=1000000,
-                                             log_file_name=log.name)
 
-            assert mylogger.hasHandlers() is True
-            assert 'RotatingFileHandler' in str(mylogger.handlers)
-            de_logger.set_stream_logging()
-            assert mylogger.debug('debug') is None
-            assert mylogger.info('infomsg') is None
+@pytest.fixture
+def log_setup(request):
```

Comment:
Introspection seems not to be used in the fixture, request is not hurting but is also not needed, better `def log_setup():`

Reply:
I have removed it.